Add default config example for ConstantName

split from #5832

update AbstractClassName to make examples for all configs shown in documentation
https://checkstyle.org/config_naming.html#ConstantName
this check need default config and example of code that is violates and expect as correct.

See original issues for examples on how to make update.

------------

PLEASE READ https://github.com/checkstyle/checkstyle/wiki/Good-practice-on-how-to-update-examples

java.lang.Error: Error was thrown while processing src\main\java\com\puppycrawl\tools\checkstyle\checks\naming\ConstantNameCheck.java
	at com.puppycrawl.tools.checkstyle.internal.XdocsJavaDocsTest.examineCheckSection(XdocsJavaDocsTest.java:209)
	at com.puppycrawl.tools.checkstyle.internal.XdocsJavaDocsTest.testAllCheckSectionJavaDocs(XdocsJavaDocsTest.java:182)
Caused by: org.opentest4j.AssertionFailedError: 
ConstantName's class-level JavaDoc ==> expected: <

@strkkk @romani I ran the command mvn clean verify.
It showed one error. How to fix it?
Failed to execute goal org.apache.maven.plugins:maven-surefire-plugin:2.22.2:test (default-test) on project checkstyle: There are test failures.

Please refer to E:\checkstyle\target\surefire-reports for the individual test results.
Please refer to dump files (if any exist) [date].dump, [date]-jvmRun[N].dump and [date].dumpstream.

[ERROR] testAllCheckSectionJavaDocs  Time elapsed: 2.84 s  <<< ERROR!
java.lang.Error: Error was thrown while processing src\main\java\com\puppycrawl\tools\checkstyle\checks\naming\ConstantNameCheck.java
	at com.puppycrawl.tools.checkstyle.internal.XdocsJavaDocsTest.examineCheckSection(XdocsJavaDocsTest.java:209)
	at com.puppycrawl.tools.checkstyle.internal.XdocsJavaDocsTest.testAllCheckSectionJavaDocs(XdocsJavaDocsTest.java:182)
Caused by: org.opentest4j.AssertionFailedError: 
ConstantName's class-level JavaDoc ==> expected: <

I ran the command mvn clean verify, It showed one error, I went through all similar PR's but couldn't find where I am going wrong.
[ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 2.259 s <<< FAILURE! - in com.puppycrawl.tools.checkstyle.internal.XdocsJavaDocsTest
[ERROR] testAllCheckSectionJavaDocs  Time elapsed: 2.259 s  <<< ERROR!
java.lang.Error: Error was thrown while processing src\main\java\com\puppycrawl\tools\checkstyle\checks\naming\ConstantNameCheck.java
	at com.puppycrawl.tools.checkstyle.internal.XdocsJavaDocsTest.examineCheckSection(XdocsJavaDocsTest.java:209)
	at com.puppycrawl.tools.checkstyle.internal.XdocsJavaDocsTest.testAllCheckSectionJavaDocs(XdocsJavaDocsTest.java:182)
Caused by: org.opentest4j.AssertionFailedError: 
ConstantName's class-level JavaDoc ==> expected: <

What shall I do to fix this error?

--- a/src/main/java/com/puppycrawl/tools/checkstyle/checks/naming/ConstantNameCheck.java
+++ b/src/main/java/com/puppycrawl/tools/checkstyle/checks/naming/ConstantNameCheck.java
@@ -64,7 +64,17 @@ import com.puppycrawl.tools.checkstyle.utils.ScopeUtil;
  * <pre>
  * &lt;module name="ConstantName"/&gt;
  * </pre>
- *
+ * <p>Example:</p>
+ * <pre>
+ * class MyClass {
+ *   public final static int FIRST_CONSTANT1 = 10; // OK
+ *   protected final static int SECOND_CONSTANT2 = 100; // OK
+ *   final static int third_Constant3 = 1000; // violation, name 'third_Constant3' must
+ *                                           // match pattern '^[A-Z][A-Z0-9]*(_[A-Z0-9]+)*$'
+ *   private final static int fourth_Const4 = 50; // violation, name 'fourth_Const4' must match
+ *                                                 // pattern '^[A-Z][A-Z0-9]*(_[A-Z0-9]+)*$'
+ * }
+ * </pre>
  * <p>
  * The following configuration apart from names allowed by default allows {@code log}
  * or {@code logger}:

--- a/src/xdocs/config_naming.xml
+++ b/src/xdocs/config_naming.xml
@@ -876,9 +876,19 @@ class MyClass5&lt;RequestT&gt; {} // OK
         <p>
           To configure the check:
         </p>
-
         <source>
 &lt;module name="ConstantName"/&gt;
+        </source>
+        <p>Example:</p>
+        <source>
+class MyClass {
+  public final static int FIRST_CONSTANT1 = 10; // OK
+  protected final static int SECOND_CONSTANT2 = 100; // OK
+  final static int third_Constant3 = 1000; // violation, name 'third_Constant3' must
+                                          // match pattern '^[A-Z][A-Z0-9]*(_[A-Z0-9]+)*$'
+  private final static int fourth_Const4 = 50; // violation, name 'fourth_Const4' must match
+                                                // pattern '^[A-Z][A-Z0-9]*(_[A-Z0-9]+)*$'
+}
         </source>
         <p>
           The following configuration apart from names allowed by default allows <code>log</code>

